packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/dataStreamSearch/workspaceSelector/workspaceSelector.tsx

```diff
@@ -34,11 +34,12 @@ export const WorkspaceSelector = ({
 
   useEffect(() => {
     OnGettingError(status === 'error');
-  }, [status, OnGettingError]);
+    if (status === 'error') setStoredWorkspace(null);
+  }, [status, OnGettingError, setStoredWorkspace]);
 
   useEffect(() => {
-    setValue('workspace', storedWorkspace || workspaceOptions[0]);
     if (workspaceOptions.length) {
+      setValue('workspace', storedWorkspace || workspaceOptions[0]);
```

Comment:
doing this has a side-effect: it sets the value as the first option `storedWorkspace` upon landing (without user manual selection)
Is that desirable?

Reply:
yes, that's the expected.  as per #2384